src/lib.rs

```diff
@@ -378,10 +387,114 @@ impl Debug for Value {
             Value::ExternFunction(slot) => write!(fmt, "(extern fn {})", slot),
             Value::Unkown => write!(fmt, "(unkown)"),
             Value::Nil => write!(fmt, "(nil)"),
+            Value::Tuple(v) => write!(fmt, "({:?})", v),
         }
     }
 }
 
+mod op {
+    use super::Value;
+    use std::rc::Rc;
+
+    fn tuple_op(a: &Rc<Vec<Value>>, b: &Rc<Vec<Value>>, f: fn (&Value, &Value) -> Value) -> Value {
+        Value::Tuple(Rc::new(a.iter().zip(b.iter()).map(|(a, b)| f(a, b)).collect()))
+    }
+
+    pub fn neg(value: &Value) -> Value {
+        match value {
+            Value::Float(a) => Value::Float(-a),
+            Value::Int(a) => Value::Int(-a),
+            Value::Tuple(a) => Value::Tuple(Rc::new(a.iter().map(neg).collect())),
+            _ => Value::Nil,
+        }
+    }
```

Comment:
This should be its own PR.

Reply:
I see where you're comming from. But without the operators extracted to functions, the Tuple operations wouldn't work since they're recursive.

Reply:
Yes, but with this it's hard to see the Tuple additions since everything's additions.